doc/build/dts/intro.rst

```diff
@@ -350,6 +328,15 @@ interrupts
 
 .. highlight:: none
 
+.. note::
+   
+   Earlier versions of Zephyr made frequent use of the ``label`` property,
+   which is distinct from the standard :ref:`node label <dt-node-labels>`.
+   Use of the label *property* and the corresponding :c:func:`DT_LABEL` macro is
+   actively discouraged in new devicetree bindings. Label properties continue to
+   persist for historical reasons in some existing bindings and overlays, but
+   should not be used in new bindings or device implementations.
+
```

Comment:
@mbolivar-nordic @galak Sensible sounding?

Reply:
DT_LABEL isn't used in bindings, and the right role is `:c:macro:`, so I would reword to:

```
Use of the label *property* in new devicetree bindings, as well as use of the :c:macro:`DT_LABEL` in new code, is actively discouraged.
```

Reply:
Fixed, thanks.